docs/architecture/architecture-overview.adoc

```diff
@@ -0,0 +1,260 @@
+         {
+             "apiVersion": "mobile.k8s.io/v1alpha1",
+             "kind": "MobileClient",
+             "metadata": {
+                 "clusterName": "",
+                 "creationTimestamp": "2018-01-04T12:35:12Z",
+                 "deletionGracePeriodSeconds": null,
+                 "deletionTimestamp": null,
+                 "labels": {
+                     "icon": "fa-android"
+                 },
+                 "name": "my.app-1515069311",
+                 "namespace": "myproject",
+                 "resourceVersion": "17980",
+                 "selfLink": "/apis/mobile.k8s.io/v1alpha1/namespaces/myproject/mobileclients/my.app-1515069311",
+                 "uid": "b5840229-f14b-11e7-95e5-b242a86257d4"
+             },
+             "spec": {
+                 "apiKey": "3e88bede-0c80-4df1-8cd9-cf9b5f50b771",
+                 "clientType": "android",
+                 "name": "my.app"
+             }
+         }
+     ],
+     "kind": "List",
+     "metadata": {
+         "resourceVersion": "",
+         "selfLink": ""
+     }
+ }
+====
+Further details on creating a custom resource object (instantiation of the CRD) is detailed https://kubernetes.io/docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions/[here].
+
+=== Provision a Mobile Service
+Another important stage in the Mobile App Dev process is to provision a Mobile value add Service to your project. This provisioning of a Mobile Service into your project is done via the Service Catalog, its not done through the MC. The MC UI extension has provided the Mobile category for the user to find the services which can be used as part of a Mobile solution.
+
+image::images/5.x-ServiceCatalog-MobileServices.png[width=1280, title="Service Catalog - Mobile Service Perspective", link="images/5.x-ServiceCatalog-MobileServices.png"]
+
+What happens behind the scene is that the 'provision' ansible playbook of the service in question is invoked. This does the provisioning of the service into the project. There is an existing https://github.com/aerogear/proposals/blob/master/apbs/create-secret-and-configmap-during-provision.md[proposal] around this topic.
+
+As part of this provisioning two objects are also created, a configMap and a secret for the new service in OpenShift, see Figure below.
+
+image::images/5.x-Architecture-MC-ServiceInstantiation.png[width=1280, title="Instantiated/Provisioned Mobile Service", link="images/5.x-Architecture-MC-ServiceInstantiation.png"]
+
+The ConfigMap which is created contains public info about the Mobile Service. The info conatined in this ConfigMap is necessary for a remote/real Mobile Client/App to be able to interact with and consume the Mobile Service. It will be most likely be the Core Client SDK which will provide the Mobile Client with this configuration (https://github.com/aerogear/proposals/pull/9/files[SDK Proposal]). The Secret that is created holds private info about the service, e.g. uname and pwd.
```

Comment:
The CLI will retrieve the configuration and the developer makes it available to the SDK via their project configuration